### ISSUE TITLE
EEA foreign key constraint violation on the entity_execution_lock table

### ISSUE THREAD

**Issue by rdimitrov** [stacklok/minder#1832]
We hit a foreign key constraint violation on the entity_execution_lock table. 

Details:
```
{
  "time": "2023-12-06T16:47:37.46839796Z",
  "stream": "stdout",
  "_p": "F",
  "level": "error",
  "component": "watermill",
  "exception.message": "error aggregating event: error locking: pq: insert or update on table \"entity_execution_lock\" violates foreign key constraint \"entity_execution_lock_repository_id_fkey\"",
  "Timestamp": 1701881257468328000,
  "message": "Handler returned error",
  "kubernetes": {
    "pod_name": "minder-5747bd877f-cfx4l",
    "namespace_name": "mediator",
    "pod_id": "c382f42c-e898-479a-b600-5b32adf3c243",
    "labels": {
      "app": "minder",
      "pod-template-hash": "5747bd877f"
    },
    "annotations": {
      "prometheus.io/path": "/metrics",
      "prometheus.io/port": "9090",
      "prometheus.io/scrape": "true"
    },
    "host": "ip-10-0-2-208.ec2.internal",
    "container_name": "minder",
    "docker_id": "8dfd862a26820bf2506c1deae9c135a2c28826ebfd4a6e335118648df3960ada",
    "container_hash": "ghcr.io/stacklok/minder/server@sha256:c6178c95fd3ff3e7c00f1526ece2f84cf6a3f8bd45860461feb0443f99f384e6",
    "container_image": "sha256:c7f6ea154e54a2bd9006c2fd98a3e0685a322e4a47fca378a032b1f24a77f163"
  }
}
```

```
{
  "time": "2023-12-06T16:51:46.668413954Z",
  "stream": "stdout",
  "_p": "F",
  "level": "info",
  "event": "c25bcaa9-d039-4d02-9c45-0de902d1d4f7",
  "entity": "pull_request",
  "repository_id": "d138b64c-c7b9-468d-9f73-c7137297855c",
  "pull_request_id": "d9855a31-4ec2-48e5-a46d-c0dde93fed81",
  "exception.message": "pq: insert or update on table \"entity_execution_lock\" violates foreign key constraint \"entity_execution_lock_repository_id_fkey\"",
  "Timestamp": 1701881506668342300,
  "message": "error locking event",
  "kubernetes": {
    "pod_name": "minder-5747bd877f-zzrp8",
    "namespace_name": "mediator",
    "pod_id": "97dcb2ef-4ab8-4f48-969e-08cae88c8682",
    "labels": {
      "app": "minder",
      "pod-template-hash": "5747bd877f"
    },
    "annotations": {
      "prometheus.io/path": "/metrics",
      "prometheus.io/port": "9090",
      "prometheus.io/scrape": "true"
    },
    "host": "ip-10-0-2-55.ec2.internal",
    "container_name": "minder",
    "docker_id": "4b5942f1913e2b02a65543580843c97aa2ccd5a5433127ec74a42c80059c5009",
    "container_hash": "ghcr.io/stacklok/minder/server@sha256:c6178c95fd3ff3e7c00f1526ece2f84cf6a3f8bd45860461feb0443f99f384e6",
    "container_image": "sha256:c7f6ea154e54a2bd9006c2fd98a3e0685a322e4a47fca378a032b1f24a77f163"
  }
}
```

**Comment by eleftherias** [comment]
The event in question was first published at 2023-12-06 13:21:08.782 UTC time
```
    {
        "@timestamp": "2023-12-06 13:21:08.782",
        "@message": {
            "time": "2023-12-06T13:21:08.782630105Z",
            "stream": "stdout",
            "_p": "F",
            "level": "info",
            "component": "watermill",
            "message_uuid": "c25bcaa9-d039-4d02-9c45-0de902d1d4f7",
            "topic": "execute.entity.event",
            "handler": "github.com/stacklok/minder/internal/controlplane.(*Server).StartHTTPServer.(*Server).HandleGitHubWebHook.func5",
            "Timestamp": 1701868868782533253,
            "message": "Publishing messages",
            "kubernetes": {
                "pod_name": "minder-bd5f8df75-6jnfp",
                "namespace_name": "mediator",
                "pod_id": "d96485f5-bd8b-4391-8ca3-af31a3c57e54",
                "labels": {
                    "app": "minder",
                    "pod-template-hash": "bd5f8df75"
                },
                "annotations": {
                    "prometheus.io/path": "/metrics",
                    "prometheus.io/port": "9090",
                    "prometheus.io/scrape": "true"
                },
                "host": "ip-10-0-2-55.ec2.internal",
                "container_name": "minder",
                "docker_id": "aba098f8aa36b38764c7128fd0ff78daf7f6d734559a5500dd7d2cd1e879f565",
                "container_hash": "ghcr.io/stacklok/minder/server@sha256:384a40a12556b4205f50e3b752d0461f44c41a8a8c8186de83b2aef126e96abd",
                "container_image": "sha256:aba5c22ac4e6000ae897446f4e5ed6a8e7db170b0155159df9f4613c1d349066"
            }
        },
        "@logStream": "logs-stacklok-eks-kube.var.log.containers.minder-bd5f8df75-6jnfp_mediator_minder-aba098f8aa36b38764c7128fd0ff78daf7f6d734559a5500dd7d2cd1e879f565.log",
        "@log": "781189302813:staging-k8s-logs"
    },
```

**Comment by evankanderson** [comment]
Fixed by #1834 and #1835
